### PR TITLE
Update manifests for argo 2.4.3

### PR DESCRIPTION
**Pull request by consideRatio**
These updates represents those mentioned in #625 and others identified
in need of updating to reflect the changes since argo 2.3.0. Argo 2.3.0
was released May 21, 2019. Argo 2.4.3 was released December 6, 2019.

By looking at the argo projects [manifests/cluster-install folder](https://github.com/argoproj/argo/tree/master/manifests/cluster-install)
and looking at the changes in the repository I tracked down all changes
and updated kubeflow/manifests to reflect those changes.

## Considerations
- I considered that the label `app: argo` is used in the kubeflow
manifests, so I ensured what I added also got that label.
- I noted CRD had `spec.names.singular` and `spec.names.listKind`
specified as the kubernetes default value while the argo manifests
didn't, so I removed the explicit default values to not deviate from
argo's manifests more than needed.
- I applied alphabetical ordering of fields in resources to mimic what I
found in existing kubeflow argo manifests.

## Resolves
- Closes issue #625
- Closes pr #638 that is simply a fix of an issue part of this PRs more
extensive fixes to keep up to date with argo.

## References
- https://github.com/argoproj/argo/releases/tag/v2.3.0
- https://github.com/argoproj/argo/releases/tag/v2.4.0
- https://github.com/argoproj/argo/releases/tag/v2.4.1
- https://github.com/argoproj/argo/releases/tag/v2.4.2
- https://github.com/argoproj/argo/releases/tag/v2.4.3

## Checklist
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate`
    3. `make test`

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/manifests/642)

### CODE REVIEW

File: argo/base/cluster-role.yaml [hunk 1]
```diff
@@ -37,12 +38,30 @@ rules:
...
   resources:
   - workflows
+  - workflows/finalizers
   verbs:
   - get
   - list
   - watch
   - update
   - patch
+  - delete
```
Comment (krishnadurai) [inline argo/base/cluster-role.yaml]: It is strange that delete was not included in the earlier manifests. I would want to verify if this has some context from a pipelines team member.

/cc @gaoning777 @Ark-kun @IronPan

File: argo/base/aggregate-roles.yaml [hunk 2]
```diff
@@ -0,0 +1,71 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply argo/base/aggregate-roles.yaml]: Moved!

File: argo/base/crd.yaml [hunk 3]
```diff
@@ -6,10 +6,22 @@ spec:
   group: argoproj.io
   names:
     kind: Workflow
-    listKind: WorkflowList
     plural: workflows
     shortNames:
     - wf
-    singular: workflow
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply argo/base/crd.yaml]: I reverted to leave these CRD fields as they were!

File: argo/base/crd.yaml [hunk 4]
```diff
@@ -6,10 +6,22 @@ spec:
   group: argoproj.io
   names:
     kind: Workflow
-    listKind: WorkflowList
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply argo/base/crd.yaml]: This CRD that register the Workflow resource kind with the k8s api-server, will in the view of the k8s api-server remain the same, as these explicit values is the default values the api-server will fall back to according to k8s documentation.

But removing them in this CRD resource would require modification of existing CRD resources which may add complexity that we could avoid!

I added back the explicit declaration of `listKind` and the `singular`!

**Comment by consideRatio** [comment]
@krishnadurai I bumped to argo 2.4.3 that was released some hours ago that included bugfixes. No change to the manifests other than the version was introduced.

**Comment by consideRatio** [comment]
@krishnadurai thanks for reviewing this =D

It is not tested in conjunction with KF Pipeline =/

I'm not yet so familiar with kubeflow ecosystem of git repos and how they relate. I'm hoping to learn more! My colleague @nikenano is bravely leading the way to adopt kubeflow and I hope we together could figure it out together and test this!

**Comment by consideRatio** [comment]
I rebased to handle merge conflicts.

**Comment by consideRatio** [comment]
I think this will go stale if it depends on me testing this e2e with kubeflow :/ i dont have the time to invest, at least not without a clear method description on how to do it.

- is there docs on how to do e2e testing?
- is there automation for that? I recall i saw an issue being fixed about e2e tests being re-enabled in the repo

**Comment by NikeNano** [comment]
I am happy to help @consideRatio with the testing, but also need some pointers on how it is expected to be tested :) 

Who are part of the pipeline team? 

Could you point us in the right direction @Ark-kun? 

Thanks for the help!

**Comment by Ark-kun** [comment]
Related: https://github.com/kubeflow/pipelines/pull/2419

**Comment by Gsantomaggio** [comment]
would be great to update to 2.5.0 because of https://github.com/kubeflow/manifests/issues/1099
Argo 2.4.3 does not fix the `pns` error. 
Does it make sense?

**Comment by saschagrunert** [comment]
Hey @consideRatio, :wave: are you still working on this one? If you don't have the time to update that PR I'd be happy to take it over. :)

**Comment by consideRatio** [comment]
@saschagrunert absolutely, closing/modifying current pr/copying etc is all okay!

**Comment by Ark-kun** [comment]
In KFP we're exploring upgrade to Argo 2.7.* : https://github.com/kubeflow/pipelines/pull/3537

**Comment by saschagrunert** [comment]
Takeover PR can be found here: https://github.com/kubeflow/manifests/pull/1145 

@consideRatio feel free to close this one if there is nothing to to here. :)

**Pull request closed without merging**